### PR TITLE
Rag agent dev

### PR DESCRIPTION
**Pull request by ryan-detect-dot-dev**
Example PR to walk through the change and review the code.

### CODE REVIEW

File: .pre-commit-config.yaml
Comment (ryan-detect-dot-dev) [inline .pre-commit-config.yaml:1]: Reminder to revert this change before opening the msticpy PR.

File: docs/notebooks/Azure_IPython_Workspace.ipynb
Comment (ryan-detect-dot-dev) [inline docs/notebooks/Azure_IPython_Workspace.ipynb:1]: Remove from PR, replaced by RagAgent.ipynb

File: msticpy/aiagents/__init__.py
Comment (ryan-detect-dot-dev) [inline msticpy/aiagents/__init__.py:1]: Refactor the config into the msticpyconfig.yaml

Comment (ryan-detect-dot-dev) [inline msticpy/aiagents/__init__.py:1]: Figure out if/how to leverage existing azure auth components, rather than just the default azure credential

File: msticpy/aiagents/assistant_agent.py
Comment (ryan-detect-dot-dev) [inline msticpy/aiagents/assistant_agent.py:1]: The key question is "what does this class do for the user?"

Revisit after the AOAI configuration changes to see if this will be worth keeping, otherwise refactor out.

File: msticpy/aiagents/rag_agent.py
Comment (ryan-detect-dot-dev) [inline msticpy/aiagents/rag_agent.py:1]: Review https://docs.python.org/3/library/importlib.resources.html or alternatives to grab resource files instead of doing this manually.

File: msticpy/aiagents/rag_agent.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# pylint: disable=too-few-public-methods
+"""Implements the RAG (Retrieval-Augmented Generation) agent for MSTICpy."""
+
+from pathlib import Path
+
+from autogen.agentchat.contrib.retrieve_user_proxy_agent import RetrieveUserProxyAgent
+
+from . import config
```
Comment (ryan-detect-dot-dev) [inline msticpy/aiagents/rag_agent.py]: Refactor to use the updated msticpy config settings

File: msticpy/aiagents/rag_agent.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        """Initialize the rag_agent with a RetrieveUserProxyAgent instance."""
+        self.ragproxyagent = RetrieveUserProxyAgent(
+            name="ragproxyagent",
+            human_input_mode="NEVER",
+            max_consecutive_auto_reply=3,
+            retrieve_config={
+                "task": "qa",
+                "docs_path": rst_files,
+                "chunk_token_size": 2000,
+                "model": config["model"],
```
Comment (ryan-detect-dot-dev) [inline msticpy/aiagents/rag_agent.py]: Ditto, refactor to use new config

File: msticpy/nbtools/rag_magic.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        self.ask_magic(cell, verbose=verbose_flag)
+
+
+# Register the magic class with IPython
+def load_ipython_extension(ipython):
+    """Register the magic class with IPython."""
+    ipython.register_magics(AutogenMagic)
+
+
+# Example Usage:
```
Comment (ryan-detect-dot-dev) [inline msticpy/nbtools/rag_magic.py]: Add to class docstrings

Comment (ryan-detect-dot-dev) [reply msticpy/nbtools/rag_magic.py]: Or the `ask` function docstrings

**Review by ryan-detect-dot-dev — COMMENTED** [review]
Getting there!

File: msticpy/aiagents/rag_agent.py
Comment (EileenG02) [reply msticpy/aiagents/rag_agent.py:1]: Using `importlib.resources` didn't give an accurate file path without editing PATH environment variables, so I opted to use both `importlib.resources` and `pathlib` to at least improve the readability of the code. I don't think the solution is what you had in mind, but it looks a bit better than before.
